mitodl/odl-video-service
# Make `ENABLE_VIDEO_PERMISSIONS` affect front-end video edit form only

#### What are the relevant tickets?
Closes #474

#### What's this PR do?
- Removes all the `if settings.ENABLE_VIDEO_PERMISSIONS` statements affecting YouTube uploads and permissions, leaves it only for the front-end VideoEditForm. 

#### How should this be manually tested?
- Set `ENABLE_VIDEO_PERMISSIONS=False` in your `.env` file
- Change a video to `is_public=True` on its admin page
- In a shell, run the following:
    ````
    from cloudsync.tasks import upload_youtube_videos
    upload_youtube_videos.delay()
    ````
- After a few minutes, go to the `Youtube Videos` admin page.  There should be a `YouTubeVideo` object for your `Video`, and its `id` should not be `None` after it has completed uploading. 
- In an incognito tab, visit the video detail page.  It should play from YouTube (look for the logo at bottom right), and not require login.
- Log in and click the 'EDIT' button on the video detail page.  You should not see any options for overriding the collection permissions.
- Upload a subtitle file for the video.
- Wait a minute then go to `https://www.youtube.com/watch?v=<YoutubeVideo.id>`, there should be a `CC` button on the video toolbar that when activated shows the captions you uploaded.
- Return to the admin page for the video, and set `is_public` to `False`.  
- Check the `YouTube Videos` admin page, there should no longer be a `YouTubeVideo` object there for that `Video`
- Check `https://www.youtube.com/watch?v=<YoutubeVideo.id>` again, you should see `This video is unavailable`
- Set `ENABLE_VIDEO_PERMISSIONS=True` in your `.env` file and restart the docker containers.
- Go to the video detail page and click 'EDIT'.  You should see options for overriding the collection permissions.  If you make it public and save it, the video should be uploaded to Youtube again.
